contracts/ProtocolGovernance.sol

```diff
@@ -122,41 +112,56 @@ contract ProtocolGovernance is IProtocolGovernance, DefaultAccessControl {
     // ------------------- PUBLIC, MUTATING, GOVERNANCE, IMMEDIATE -----------------
 
     /// @inheritdoc IProtocolGovernance
-    function rollbackStagedPermissions() external {
+    function rollbackStagedGrantedPermissions() external {
         _requireAdmin();
-        require(_isStagedToCommit(), ExceptionsLibrary.INVALID_STATE);
         _clearStagedPermissions();
-        delete _permissionAddressesTimestamp;
-        emit RolledBackStagedPermissions(tx.origin, msg.sender);
+        emit RolledBackStagedGrantedPermissions(tx.origin, msg.sender);
     }
 
     /// @inheritdoc IProtocolGovernance
     function commitStagedPermissions() external {
         _requireAdmin();
-        require(_isStagedToCommit(), ExceptionsLibrary.INVALID_STATE);
-        require(block.timestamp >= _permissionAddressesTimestamp, ExceptionsLibrary.TIMESTAMP);
-        uint256 length = _stagedPermissionAddresses.length();
+        uint256 length = _stagedGrantedPermissionAddresses.length();
         for (uint256 i; i != length; ++i) {
-            address delayedAddress = _stagedPermissionAddresses.at(i);
-            uint256 delayedPermissionMask = _stagedPermissionMasks[delayedAddress];
-            _permissionMasks[delayedAddress] = delayedPermissionMask;
-            if (delayedPermissionMask == 0) {
-                _permissionAddresses.remove(delayedAddress);
-            } else {
-                _permissionAddresses.add(delayedAddress);
+            address stagedAddress = _stagedGrantedPermissionAddresses.at(i);
+            if (block.timestamp >= grantedPermissionAddressTimestamps[stagedAddress]) {
+                permissionMasks[stagedAddress] = stagedGrantedPermissionMasks[stagedAddress];
+                if (permissionMasks[stagedAddress] == 0) {
+                    _permissionAddresses.remove(stagedAddress);
+                } else {
+                    _permissionAddresses.add(stagedAddress);
+                }
             }
         }
         _clearStagedPermissions();
-        delete _permissionAddressesTimestamp;
+        delete pendingParamsTimestamp;
+    }
+
+    /// @inheritdoc IProtocolGovernance
+    function commitStagedPermission(address stagedAddress) external {
+        _requireAdmin();
+        uint256 stagedToCommitAt = grantedPermissionAddressTimestamps[stagedAddress];
+        require(block.timestamp >= stagedToCommitAt, ExceptionsLibrary.TIMESTAMP);
+        require(stagedToCommitAt != 0, ExceptionsLibrary.NULL);
+        permissionMasks[stagedAddress] |= stagedGrantedPermissionMasks[stagedAddress];
```

Comment:
Should be just =, not |= ?

Reply:
Or if you say that staged are only grant permissions, then line 128 should be changed